### PR TITLE
[ToolTip]: Add width, padding, and borderRadius

### LINKED ISSUE #7362: [Tooltip]: Allow for more padding and custom width (ability to adapt to the size of its children & accommodate longer content)
## Feature request summary

Currently, the [Tooltip](https://polaris.shopify.com/components/tooltip) has **fixed padding** and **fixed width.** The padding is quite minimal and there a couple places in web where we need a tooltip that could adapt to the size of its children and accommodate longer content than the regular [Tooltip](https://polaris.shopify.com/components/tooltip).

<details><summary>EXPAND FOR SCREENSHOTS</summary>

13-08-02yhd-xb7g8

</details>

## Rationale

Because this doesn't exist, we're relying on custom components in web, like this [Padded Tooltip](https://github.com/Shopify/web/blob/main/app/sections/Products/components/PaddedTooltip/PaddedTooltip.tsx) component based on a [Polaris Popover](https://polaris.shopify.com/components/popover) because we needed a tooltip that could adapt to the size of its children and accommodate longer content than the regular [Tooltip](https://polaris.shopify.com/components/tooltip) (the Polaris Tooltip component has a fixed width and minimal padding).

Of course, if Polaris can do this - it would be AMAZING 😄. Instead of consuming custom components which can cause a11y concerns, and could lead into inconsistency throughout the web admin 😱. 

🌟 Feature requests that are not yet planned will be closed. We then use the issue’s :+1: upvotes to track and set priorities. See the [contribution guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md) for more information.

### PR DESCRIPTION
**Pull request by ginabak**
### WHY are these changes introduced?

Currently, the [Tooltip](https://polaris.shopify.com/components/tooltip) has fixed padding and fixed width for longer content and no border radius options. 

Part of https://github.com/Shopify/inventory-states-ux/issues/636
Fixes https://github.com/Shopify/polaris/issues/7362
Related [Figma](https://www.figma.com/file/94gzO5MY58N8dVEeleOgOq/%E2%9C%A8-Inventory-States-%E2%80%93-Web?node-id=6000%3A424027&t=zxgshjdZr0lEghbl-0) 

#### What was all added to Tooltip? 

Optional Props: 
**Width** = `'default' | 'wide'`
**Padding** = `'default' | Extract<SpacingSpaceScale, '4'>` _(this has a default since there are horizontal and vertical padding involved)_ 
**BorderRadius** = `Extract<ShapeBorderRadiusScale, '1' | '2'>`

### WHAT is this pull request doing?
**TL;DR summary of changes:** 
1. Added 3 optional props: `width`, `padding`,  `borderRadius`. 
2. Modified `Tooltip` storybook to accommodate new props and removed custom CSS

Visually: 

Screenshot 2023-01-09 at 10 37 58 AM

Screenshot 2023-01-11 at 11 57 11 AM

Screenshot 2023-01-11 at 11 57 22 AM

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

[Here is the most recent chromatic storybook build!](https://5d559397bae39100201eedc1-xyomrzbunn.chromatic.com/?path=/story/all-components-tooltip--default)

For Shopi-humans, here are spin links you can test out 😄: 
🌀 [SPIN: Inventory List](https://admin.web.polaris-tool.gina-bak.us.spin.dev/store/shop1/products/inventory?location_id=1) 🌀
🌀 [SPIN: Inventory Card](https://admin.web.polaris-tool.gina-bak.us.spin.dev/store/shop1/products/1/variants/1) 🌀

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import {QuestionMarkMajor} from '@shopify/polaris-icons';
import React from 'react';

import {Page, TextContainer, Text, Tooltip, Icon} from '../src';

export function Playground() {
  const helpText = (
    <TextContainer>
      <p> TESTING 1, 2, 3 </p>
    </TextContainer>
  );
  return (
    <Page title="TOOLTIP PLAYGROUND">
      {/* Add the code you want to test in here */}
      <Tooltip
        content={helpText}
        active
        width="wide"
        padding="loose"
        borderRadius="slight"
      >
        <Icon source={QuestionMarkMajor} color="base" />
      </Tooltip>
    </Page>
  );
}

```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [x] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/rotten-hornets-move.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (LauraAubin) [inline .changeset/rotten-hornets-move.md:1]: lol what is this file title 🤔

Comment (ginabak) [reply .changeset/rotten-hornets-move.md:1]: 😆 HAHA, it's created by `yarn changeset` 😄 HAHA. Here's more context: https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#adding-a-changeset

File: polaris-react/src/components/Tooltip/Tooltip.stories.tsx [hunk 2]
```diff
@@ -1,26 +1,154 @@
...
+        </Stack>
+      </Tooltip>
+    </Stack>
+  );
+}
+
+export function Padding() {
+  return (
+    <Stack spacing="extraLoose" distribution="fill">
+      <Tooltip active content="This content has default/ tight padding">
```
Comment (LauraAubin) [inline polaris-react/src/components/Tooltip/Tooltip.stories.tsx]:
```suggestion
      <Tooltip active content="This content has tight (default) padding">
```

File: polaris-react/src/components/Tooltip/Tooltip.stories.tsx [hunk 3]
```diff
@@ -1,26 +1,154 @@
...
+    </Stack>
+  );
+}
+
+export function BorderRadius() {
+  return (
+    <Stack spacing="extraLoose" distribution="fill">
+      <Tooltip
+        active
+        content="This content has the default/ base border radius"
```
Comment (LauraAubin) [inline polaris-react/src/components/Tooltip/Tooltip.stories.tsx]:
```suggestion
        content="This content has the base (default) border radius"
```

File: polaris-react/src/components/Tooltip/Tooltip.stories.tsx (hunk 2 above)
Comment (ginabak) [reply polaris-react/src/components/Tooltip/Tooltip.stories.tsx]: changed!

File: polaris-react/src/components/Tooltip/Tooltip.stories.tsx (hunk 3 above)
Comment (ginabak) [reply polaris-react/src/components/Tooltip/Tooltip.stories.tsx]: changed!

File: polaris-react/src/components/Tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -7,6 +7,10 @@ import {useToggle} from '../../utilities/use-toggle';
 
 import {TooltipOverlay, TooltipOverlayProps} from './components';
 
+export type Width = 'default' | 'wide';
+export type Padding = 'tight' | 'loose';
+export type BorderRadius = 'none' | 'slight';
```
Comment (alex-page) [inline polaris-react/src/components/Tooltip/Tooltip.tsx]: We should move away from this naming convention and get these values to be as close to the tokens we use in our system.

File: polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss [hunk 5]
```diff
@@ -32,15 +29,38 @@ $content-max-width: 200px;
...
+}
+
+.loose {
+  padding: var(--p-space-4);
+}
+
+// Border Radius of Content
+.none {
+  border-radius: var(--p-border-radius-1);
+}
```
Comment (alex-page) [inline polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss]: This one is confusing to me `none` still has a border-radius. I would expect hard corners if the value was `none`.

Comment (ginabak) [reply polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss]: Very good point. Let me rethink the naming.

File: polaris-react/src/components/Tooltip/Tooltip.tsx (hunk 4 above)
Comment (ginabak) [reply polaris-react/src/components/Tooltip/Tooltip.tsx]: ❤️ AHH, okie, I thought so ... but I'm struggling with the values as I only want to allow for one or the other and not all tokens? I was also struggling with `width` and `padding` one. I was thinking about doing the padding like in [AlphaCard](https://github.com/Shopify/polaris/blob/128f147d19354adbb24c48443e7c47ff6dd06c74/polaris-react/src/components/AlphaCard/AlphaCard.tsx#L18), [Bleed](https://github.com/Shopify/polaris/blob/128f147d19354adbb24c48443e7c47ff6dd06c74/polaris-react/src/components/Bleed/Bleed.tsx#L12), and [Box](https://github.com/Shopify/polaris/blob/128f147d19354adbb24c48443e7c47ff6dd06c74/polaris-react/src/components/Box/Box.tsx#L51)), but I wanted to just give **two** simple options. 

Would having something like: 
```TypeScript
export type Width = 'default' | 'wide'; 
export type Padding = 'default' | '4';
export type BorderRadius = 'default' | '2'  // or we could do '1' | '2' to match the --p-border-radius-1 and --p-border-radius-2 ?
```

be better? 🤔

File: polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss (hunk 5 above)
Comment (ginabak) [reply polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss]: I changed it to `default` but would love to know how I can try to use polaris tokens for the types 🤔 ❤️

File: polaris-react/src/components/Tooltip/Tooltip.tsx (hunk 4 above)
Comment (alex-page) [reply polaris-react/src/components/Tooltip/Tooltip.tsx]: cc @aveline @laurkim if you have any thoughts based on the layout work that is happening.

Comment (aaronccasanova) [reply polaris-react/src/components/Tooltip/Tooltip.tsx]: > We should move away from this naming convention and get these values to be as close to the tokens we use in our system.

Agreed, having one off naming conventions doesn't make for a cohesive and intuitive system imo. I understand your intent @ginabak and I too would prefer a limited set of aliases (my go-to's are `small` | `medium` | `large`). However, for consistency and intuitiveness it may be worth exposing the whole token group (similar to [borderRadius](https://github.com/Shopify/polaris/blob/2bacc3bdf03d77a1f385758df2df264d0f521e6c/polaris-react/src/components/Box/Box.tsx#LL53C29-L53C29) and [padding](https://github.com/Shopify/polaris/blob/2bacc3bdf03d77a1f385758df2df264d0f521e6c/polaris-react/src/components/Box/Box.tsx#L51) in Box).

File: polaris-react/src/components/Tooltip/Tooltip.stories.tsx [hunk 6]
```diff
@@ -1,26 +1,154 @@
...
+} from '@shopify/polaris';
 
 export default {
   component: Tooltip,
 } as ComponentMeta<typeof Tooltip>;
 
 export function Default() {
   return (
-    <div style={{padding: '75px 0'}}>
+    <Stack spacing="extraLoose" distribution="fill">
```
Comment (highfieldjames) [inline polaris-react/src/components/Tooltip/Tooltip.stories.tsx]: Do we need to specify the spacing and distribution for this Stack (or do we even need a Stack) if it's only got one child component?

File: polaris-react/src/components/Tooltip/Tooltip.tsx [hunk 7]
```diff
@@ -30,6 +34,21 @@ export interface TooltipProps {
...
+   * @default 'default'
+   */
+  width?: Width;
+  /**
+   * Padding of content
+   * @default 'default'
+   */
+  padding?: Padding;
+  /**
+   * Border Radius of tooltip
```
Comment (highfieldjames) [inline polaris-react/src/components/Tooltip/Tooltip.tsx]:
```suggestion
   * Border radius of the tooltip
```

**Review by highfieldjames — APPROVED** [review]
Looks great, thanks Gina! Love the new border radius option 😍

File: polaris-react/src/components/Tooltip/Tooltip.tsx (hunk 4 above)
Comment (ginabak) [reply polaris-react/src/components/Tooltip/Tooltip.tsx]: hi @alex-page ❤️ I think both Aveline and Lo are out of office - is there anyone else we could ping? 😄

File: polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss [hunk 8]
```diff
@@ -32,15 +29,38 @@ $content-max-width: 200px;
...
+// Padding of Content
+.default {
+  padding: var(--p-space-1) var(--p-space-2);
+}
+
+.loose {
+  padding: var(--p-space-4);
+}
+
+// Border Radius of Content
```
Comment (LauraAubin) [inline polaris-react/src/components/Tooltip/components/TooltipOverlay/TooltipOverlay.scss]:
```suggestion
// Border radius of Content
```

File: polaris-react/src/components/Tooltip/Tooltip.tsx (hunk 4 above)
Comment (ginabak) [reply polaris-react/src/components/Tooltip/Tooltip.tsx]: cc @kyledurand 😄

Comment (kyledurand) [reply polaris-react/src/components/Tooltip/Tooltip.tsx]: 👋 Can we do something like this?

```tsx
export type Padding = 'default' | Extract<SpacingSpaceScale, '4'>;
export type BorderRadius = Extract<ShapeBorderRadiusScale, '1' | '2'>;
```
It's awkward because you need the 'default' key since the container has different values for vertical / horizontal padding.

Or you could break down the padding into vertical / horizontal properties:
```tsx
export interface Padding {
  vertical?: Extract<SpacingSpaceScale, '1' | '2' | '4'>;
  horizontal?: Extract<SpacingSpaceScale, '1' | '2' | '4'>;
}
```

I'm worried the second would lead to more inconsistencies though so I'd opt for keeping the default key for now. I'm down to pair on this if you want a hand.

File: polaris-react/src/components/Tooltip/Tooltip.tsx [hunk 9]
```diff
@@ -7,6 +7,10 @@ import {useToggle} from '../../utilities/use-toggle';
 
 import {TooltipOverlay, TooltipOverlayProps} from './components';
 
+export type Width = 'default' | 'wide';
+export type Padding = 'default' | 'loose';
+export type BorderRadius = 'default' | 'slight';
```
Comment (chloerice) [inline polaris-react/src/components/Tooltip/Tooltip.tsx:12]:
```suggestion
export type BorderRadius = 'default' | 'large';
```

Since the "slight" is an increase in border-radius from the default, it might be clearer named something like "large"

File: polaris-react/src/components/Tooltip/Tooltip.stories.tsx (hunk 6 above)
Comment (chloerice) [reply polaris-react/src/components/Tooltip/Tooltip.stories.tsx]: If the `Stack` is just for adding spacing, we could use `<Box padding="8">` ('8' = --p-space-8 = 32px = 'extraLoose' )

File: polaris-react/src/components/Tooltip/Tooltip.tsx (hunk 4 above)
Comment (ginabak) [reply polaris-react/src/components/Tooltip/Tooltip.tsx]: you are all amazing!! THANK YOU ❤️

**Comment by tracytees** [comment]
Looks good from a design perspective~ Not sure if @kyledurand has anything extra to add though :)

File: polaris-react/src/components/Tooltip/Tooltip.tsx [hunk 10]
```diff
@@ -7,6 +11,10 @@ import {useToggle} from '../../utilities/use-toggle';
 
 import {TooltipOverlay, TooltipOverlayProps} from './components';
 
+export type Width = 'default' | 'wide';
+export type Padding = 'default' | Extract<SpacingSpaceScale, '4'>;
+export type BorderRadius = Extract<ShapeBorderRadiusScale, '1' | '2'>;
```
Comment (chloerice) [inline polaris-react/src/components/Tooltip/Tooltip.tsx:16]:
```suggestion
export type BorderRadius = 'default' | Extract<ShapeBorderRadiusScale, '2'>;
```

File: polaris-react/src/components/Tooltip/Tooltip.stories.tsx [hunk 11]
```diff
@@ -1,26 +1,152 @@
...
+          <Icon source={QuestionMarkMinor} color="base" />
+        </Stack>
+      </Tooltip>
+    </Stack>
   );
 }
 
 export function VisibleOnlyWithChildInteraction() {
   return (
-    <div style={{width: '200px'}}>
```
Comment (chloerice) [inline polaris-react/src/components/Tooltip/Tooltip.stories.tsx]: Let's keep the limited width here so the segmented button group doesn't stretch to the page width

**Review by chloerice — APPROVED** [review]

File: polaris-react/src/components/Tooltip/Tooltip.tsx (hunk 10 above)
Comment (ginabak) [reply polaris-react/src/components/Tooltip/Tooltip.tsx:16]: (spoke on slack 😄): leaving without the `default` on `BorderRadius` so it matches the tokens. Kept `default` for padding as there is both horizontal and vertical ❤️

**Comment by ginabak** [comment]
Shipping as the a11y contrast error is unrelated to this PR! 🚢
